src/workerd/api/basics.c++

```diff
@@ -669,6 +712,97 @@ void AbortSignal::triggerAbort(
   dispatchEventImpl(js, js.alloc<Event>(kj::str("abort")));
 }
 
+void AbortSignal::serialize(jsg::Lock& js, jsg::Serializer& serializer) {
+  auto& handler = JSG_REQUIRE_NONNULL(serializer.getExternalHandler(), DOMDataCloneError,
+      "AbortSignal can only be serialized for RPC.");
+  auto externalHandler = dynamic_cast<RpcSerializerExternalHander*>(&handler);
+  JSG_REQUIRE(
+      externalHandler != nullptr, DOMDataCloneError, "AbortSignal can only be serialized for RPC.");
+
+  serializer.writeRawUint32(static_cast<uint>(canceler->isCanceled()));
+  serializer.writeRawUint32(static_cast<uint>(flag));
+  KJ_IF_SOME(r, reason) {
+    serializer.write(js, r.getHandle(js));
+  } else {
+    serializer.write(js, js.undefined());
+  }
+
+  auto& ioContext = IoContext::current();
+
+  auto streamCap =
+      externalHandler->writeStream([&](rpc::JsValue::External::Builder builder) mutable {
+    auto field = builder.initAbortSignal();
+    field.setSignal(kj::heap<AbortSignalImpl>(ioContext, JSG_THIS));
+  });
+
+  // Keep track of every AbortSignal cloned from this one.
+  // If this->triggerAbort(...) is called, each rpcClient will be informed.
+  rpcClients.add(kj::mv(streamCap).castAs<rpc::AbortSignal>());
+}
+
+jsg::Ref<AbortSignal> AbortSignal::deserialize(
+    jsg::Lock& js, rpc::SerializationTag tag, jsg::Deserializer& deserializer) {
+  auto& handler = KJ_REQUIRE_NONNULL(
+      deserializer.getExternalHandler(), "got AbortSignal on non-RPC serialized object?");
+  auto externalHandler = dynamic_cast<RpcDeserializerExternalHander*>(&handler);
+  KJ_REQUIRE(externalHandler != nullptr, "got AbortSignal on non-RPC serialized object?");
+
+  auto isCanceled = static_cast<bool>(deserializer.readRawUint32());
+  auto flag = static_cast<Flag>(deserializer.readRawUint32());
+  auto reason = deserializer.readValue(js);
+
+  auto reader = externalHandler->read();
```

Comment:
The `reader` contains an `AbortSignal` capability but it appears you don't ever use it here? You just drop it?

Reply:
Safe to say I have no idea what I'm doing. Please let me know if there's a better way to encode the AbortSignal interface in terms of RPC.

Reply:
AbortSignals only go one way, there's no need for the cloned signal to send anything back to its parent.

Reply:
Actually hold on, back up.

I was assuming the idea was that, once an `AbortSignal` has been sent over the wire, then either end can listen on the signal _and_ either end can _trigger_ the signal, so both ends are listening on each other.

But actually does that make sense? An `AbortSignal` JS object lets you _receive_ the signal. It doesn't let you trigger it -- that's what `AbortController` is for.

So you actually _only_ want to use a stream slot here (pointing caller -> callee)? You don't need any callee -> caller capability at all?

Reply:
Correct